Client/Frontend/Login Management/LoginListViewController.swift

```diff
@@ -166,6 +172,44 @@ class LoginListViewController: UIViewController {
     }
 }
 
+// MARK: - Favicon Loading
+extension LoginListViewController {
+
+    private func loadFaviconsForLogins(logins: [Login]) {
+        logins.forEach { login in
+            loadFaviconsForLoginFromHistory(login).uponQueue(dispatch_get_main_queue()) { result in
```

Comment:
Rework this to bulk query. If we're showing 100 logins, we're going to trigger 100 different queries and ~200 different runnables with 100 different UI updates. At the very least we can grab `(url, icon)` pairs, cluster them (see my `Array.groupBy` commit), and skip straight to updating the UI.